internal/generator/generator_util.go

```diff
@@ -201,3 +228,52 @@ func genFromSrcCode(codePath string, mapVariableName string, matchFnName string,
 
 	return nil
 }
+
+func strList2Map(strs []string) map[string]struct{} {
+	if strs == nil {
+		return nil
+	}
+
+	rv := map[string]struct{}{}
+	for _, str := range strs {
+		rv[str] = struct{}{}
+	}
+
+	return rv
+}
+
+//nolint:nilnil, mnd
+func getOverride(rawInputs []string) (map[string][]mask, error) {
+	if rawInputs == nil {
+		return nil, nil
+	}
+
+	override := map[string][]mask{}
+
+	for _, rawStr := range rawInputs {
+		// rawStr should follow the format: directive:bitmask00|bitmask01|...,bitmask10|bitmask11|...
+		subs := strings.Split(rawStr, ":")
+		if len(subs) != 2 {
+			return nil, fmt.Errorf("override parameter %s format wrong", rawStr)
+		}
+
+		directive := strings.TrimSpace(subs[0])
+		if directive == "" {
+			return nil, errors.New("directive name can't be empty")
+		}
+
+		if strings.TrimSpace(subs[1]) == "" {
+			return nil, errors.New("directive masks can't be empty")
+		}
+
+		for _, varNames := range strings.Split(subs[1], ",") {
+			directiveMask := mask{}
+			for _, varName := range strings.Split(varNames, "|") {
+				directiveMask = append(directiveMask, strings.TrimSpace(varName))
+			}
```

Comment:
Nit: If you know the size of the slice needed, it's better to set it. It doesn't really matter that much since this is a tool that is only run occasionally but good practice anyway.

```go
names := strings.Split(varNames, "|")
if len(names) == 0 {
	continue
}

directiveMask := make(mask, len(names))
for i, name := range names {
	directiveMask[i] = strings.TrimSpace(name)
}
```

Reply:
> Nit: If you know the size of the slice needed, it's better to set it.

Get it! Applied

Reply:
Applied: fix the length of directiveMask, add one error and corresponding ut